metrics/metrics.go

```diff
@@ -106,7 +117,33 @@ func (t *connTracer) StartedConnection(local, _ net.Addr, _ logging.VersionNumbe
 	)
 }
 
-func (t *connTracer) ClosedConnection(logging.CloseReason)                     {}
+func (t *connTracer) ClosedConnection(r logging.CloseReason) {
+	var tags []tag.Mutator
```

Comment:
I wonder whether this could be simplified by doing more of this processing during the CloseReason creation already. For example the type of reason, or whether it's remote or not could be set when creating the `CloseReason`, rather than interpreting errors later on. Essentially, make `CloseReason` POD?

Reply:
Yes, `CloseReason` is a mess. I'd like to clean this up, but I'm not really sure how.